src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/PushHookProcessor.java

```diff
@@ -44,10 +68,124 @@ public void process(String payload, BitbucketType instanceType) {
             }
             if (push != null) {
                 String owner = push.getRepository().getOwnerName();
-                String repository = push.getRepository().getRepositoryName();
+                final String repository = push.getRepository().getRepositoryName();
+                if (push.getChanges().isEmpty()) {
+                    LOGGER.log(Level.INFO, "Received hook from Bitbucket. Processing push event on {0}/{1}",
+                            new Object[]{owner, repository});
+                    scmSourceReIndex(owner, repository);
+                } else {
+                    SCMHeadEvent.Type type = null;
+                    for (BitbucketPushEvent.Change change: push.getChanges()) {
+                        if ((type == null || type == SCMEvent.Type.CREATED) && change.isCreated()) {
```

Comment:
Nope, if there are many changes and they are all created changes, then type is created.
If there are many changes and they are all removed changes, then type is removed.
If there are any update changes or a mix of change types, then type is updated